src/StreamJsonRpc/StreamJsonRpc.csproj

```diff
@@ -3,8 +3,10 @@
     <TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-
+    <ErrorLog>$(MSBuildProjectFullPath).$([System.Guid]::NewGuid().ToString().Substring(0,8)).sarif</ErrorLog>
```

Comment:
What reads this in order to produce the files?
Can we place it in the `$(IntermediateOutputPath)` instead of in the source directory to preserve our read-only source tree?
If this should be applied to all non-test projects, can we promote it to the `src/Directory.Build.props` file?

Reply:
```
 ### Copy existing Roslyn Analyzer logs to be analyzed
- task: RoslynAnalyzers@3
  displayName: Run Roslyn Analyzers.
  inputs:
    copyLogsOnly: 'true'
  continueOnError: True
```
at line 31 will read this file.

I will try moving the file and see if it works but the docs for the `RoslynAnalyzers` task specify that this is what should be used.

Reply:
Removed as we rely on the build breaking to avoid Roslyn Analyzers issues